### PR TITLE
RaoPtdfParameters small refactor

### PR DESCRIPTION
**Pull request by pet-mit**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Small refactor fr RaoPtdfParameters

**What is the current behavior?** *(You can also link to an open issue here)*
In RaoPtdfParameters, the boundaries attribute is null by default. This can cause a crash when exporting RaoParameters to json format.
On another hand, "ptdf-sum-lower-bound" which is used only to bound PTDFs in RAO is a parameter in RaoParameters 

**What is the new behavior (if this is a feature change)?**
The default boundaries is an empty list, so no bug on export
The "ptdf-sum-lower-bound" parameters has been moved into the RaoPtdfParameters to be more consistent

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
Users should set "ptdf-sum-lower-bound" in the RaoPtdfParameters extension and not in RaoParameters anymore

### CODE REVIEW

File: ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoUtil.java [hunk 1]
```diff
@@ -64,10 +64,8 @@ public static void checkParameters(RaoParameters raoParameters, RaoInput raoInpu
             if (raoInput.getGlskProvider() == null) {
                 throw new FaraoException("Relative margin objective function requires a GLSK provider.");
             }
-            if (Objects.isNull(raoParameters.getExtension(RaoPtdfParameters.class))
-                || Objects.isNull(raoParameters.getExtension(RaoPtdfParameters.class).getBoundaries())
-                || raoParameters.getExtension(RaoPtdfParameters.class).getBoundaries().isEmpty()) {
-                throw new FaraoException("Relative margin objective function requires a list of pairs of country boundaries.");
+            if (raoParameters.getPtdfBoundaries().isEmpty()) {
+                throw new FaraoException("Relative margin objective function requires a list of pairs of country boundaries to compute PTDF sums upon.");
```
Comment (TheMaskedTurtle) [inline ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoUtil.java:68]: Below that there is 
`if ((raoParameters.isRaoWithLoopFlowLimitation()
            || raoParameters.getObjectiveFunction().doesRequirePtdf())
            && (raoInput.getReferenceProgram() == null)) {`
Is it normal for the doesRequirePtdf part ?

File: ra-optimisation/rao-commons/src/test/java/com/farao_community/farao/rao_commons/RaoUtilTest.java [hunk 2]
```diff
@@ -59,11 +59,9 @@ public void setUp() {
     }
 
     private void addPtdfParameters(List<String> boundaries) {
-        RaoPtdfParameters raoPtdfParameters = new RaoPtdfParameters();
         if (boundaries != null) {
-            raoPtdfParameters.setBoundariesFromCountryCodes(boundaries);
+            raoParameters.setPtdfBoundariesFromCountryCodes(boundaries);
         }
-        raoParameters.addExtension(RaoPtdfParameters.class, raoPtdfParameters);
     }
```
Comment (TheMaskedTurtle) [inline ra-optimisation/rao-commons/src/test/java/com/farao_community/farao/rao_commons/RaoUtilTest.java:65]: Can you just remove the add method when the argument is null ?

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoUtil.java (hunk 1 above)
Comment (pet-mit) [reply ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoUtil.java:68]: it's to merge the case with the isRaoWithLoopFlowLimitation case, when requiring a refprog (if i understand the question well)

File: ra-optimisation/rao-commons/src/test/java/com/farao_community/farao/rao_commons/RaoUtilTest.java (hunk 2 above)
Comment (pet-mit) [reply ra-optimisation/rao-commons/src/test/java/com/farao_community/farao/rao_commons/RaoUtilTest.java]: done

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**